src/tests/class_tests/openms/source/ElementDB_test.cpp

```diff
@@ -135,6 +152,70 @@ START_SECTION(void addElement(const std::string& name,
   // ptr addresses cannot change, otherwise we are in trouble since EmpiricalFormula uses those
   TEST_EQUAL(oxygen, new_oxygen)
   TEST_REAL_SIMILAR(oxygen->getAverageWeight(), 16.8994405) // average weight has changed
+
+  TEST_EQUAL(e_ptr->getElement(800), elem_nullPointer)
+  e_ptr->addElement("NewElement", "NE", 800u, oxygen_abundance, oxygen_mass, false);
+  const Element * new_ele = e_ptr->getElement(800);
+  TEST_REAL_SIMILAR(new_ele->getAverageWeight(), 16.8994405) // average weight of new element
+
+  // Test that we cannot add elements twice with replace=false 
+  TEST_EXCEPTION(Exception::IllegalArgument, e_ptr->addElement("NewElement", "NE", 800u, oxygen_abundance, oxygen_mass, false))
+
+  // Test that we can add elements twice with replace=true 
+  e_ptr->addElement("NewElement", "NE", 800u, oxygen_abundance, oxygen_mass, true);
+}
+END_SECTION
+
+START_SECTION( void addIsotope(const std::string& name, const std::string& symbol, const unsigned int an, double abundance, double mass, double half_life, Isotope::DecayMode decay, bool replace_existing))
+{
+  const Isotope * iso1 = e_ptr->getIsotope("(238)U");
+  e_ptr->addIsotope("Uranium", "U", 92u, 1.3, 238.05, 1e5, Isotope::DecayMode::UNKNOWN, true);
+
+  const Isotope * iso2 = e_ptr->getIsotope("(238)U");
+  // ptr addresses cannot change, otherwise we are in trouble since EmpiricalFormula uses those
+  TEST_EQUAL(iso1, iso2)
+  TEST_REAL_SIMILAR(iso1->getAbundance(), 1.3) // natural abundance has changed
+
+  // we have now managed to have 130% natural abundance for Uranium
+  // NOTE: this is a major problem for average weight calculations etc
+  const Element* element = e_ptr->getElement(92);
+  double sum = 0;
+  for (auto& iso : element->getIsotopeDistribution()) {sum += iso.getIntensity();}
+  TEST_REAL_SIMILAR(sum, 1.30725795222315);
+
+  // new Uranium isotope added
+  TEST_EQUAL(e_ptr->getIsotope("(314)C") == nullptr, true)
+  int nr_isotopes = element->getIsotopes().size();
+  e_ptr->addIsotope("Uranium", "U", 92u, 0.3, 314.0, 1e5, Isotope::DecayMode::UNKNOWN, false);
```

Comment:
should adding a new isotope with a natural abundance only work if DecayMode == NONE?

Reply:
yes, that would be a good idea

Reply:
actually thinking about it: I dont think so, you can clearly do MS measurements on molecules that have radioactive isotopes incorporated, such as Uranium and C14. There are samples that have a specific and well known composition of radioactive elements and maybe the user wants to model this. This is also where `addIsotope` would come in, if the user wants to adjust parameters to a sample they have at hand

Reply:
ok, fair point. But then the documentation should not mention `natural` abundance.
https://github.com/OpenMS/OpenMS/pull/5956/files#diff-8b049d8c7ee5d2d85f2741157e4f6ae5cf91ff98c2247f000e5477f3f3389569R137
https://github.com/OpenMS/OpenMS/pull/5956/files#diff-db97d6be7dd38c6a135f55d80d9f03eb2bcea2d0577d5a449845a0f05821811eR51
https://github.com/OpenMS/OpenMS/pull/5956/files#diff-db97d6be7dd38c6a135f55d80d9f03eb2bcea2d0577d5a449845a0f05821811eR95